### PR TITLE
feat(formatting): implement separate fiat currency formatting for value and price

### PR DESCRIPTION
**Pull request by devpramoth**
https://linear.app/safe-global/issue/GRO-130/implement-separate-fiat-currency-formatting-for-values-and-prices

## Summary

Implements separate formatting rules for fiat currency values (total balances) and prices (unit prices) to improve readability and consistency.

### Value Formatting (Total Balances)
- Always display 2 decimal places for values ≥ $0.01
- Display `<$0.01` threshold indicator for values < $0.01
- Display `$0.00` for zero values

### Price Formatting (Unit Prices)
- Values ≥ $0.01: 2 decimal places
- Values $0.0001 - $0.0099: 4-6 decimal places (adaptive based on magnitude)
- Values < $0.0001: 6 decimal places or threshold indicator

### Additional Improvements
- Removed compact notation (K, M, B suffixes) - always display full numbers
- Removed hover tooltips from FiatValue component
- Updated AssetsTable price column to use price mode for accurate unit price display

## Technical Details

- Added `mode` parameter to `formatCurrency` function (`'value' | 'price'`, defaults to `'price'` for backward compatibility)
- Updated `FiatValue` component to accept optional `mode` prop (defaults to `'value'`)
- All existing usages continue to work with backward compatibility

## Testing

- ✅ All unit tests passing (35/35 in formatNumber, 6/6 in FiatValue)
- ✅ Type checking passes
- ✅ No linting errors
- ✅ Backward compatible

## Files Changed

- `packages/utils/src/utils/formatNumber.ts` - Core formatting logic
- `packages/utils/src/utils/__tests__/formatNumber.test.ts` - Updated tests
- `apps/web/src/components/common/FiatValue/index.tsx` - Added mode prop, removed tooltip
- `apps/web/src/components/common/FiatValue/FiatValue.test.tsx` - Updated tests
- `apps/web/src/components/balances/AssetsTable/index.tsx` - Use price mode for price column

---

> [!NOTE]
> Introduces separate fiat formatting modes for values vs prices, updates FiatValue and AssetsTable to use them, removes tooltips/compact notation, and revises tests.
> 
> - **Utils (`packages/utils`)**
>   - **`formatCurrency`**: Adds `mode` (`'value' | 'price'`, default `'price'`); implements fixed 2-decimals/threshold for value mode and adaptive precision for price mode; removes compact (K/M/B) notation; preserves narrow space formatting.
>   - **Tests**: Expand coverage for both modes, thresholds, large numbers, and edge cases.
> - **Web Components (`apps/web`)**
>   - **`FiatValue`**: Accepts optional `mode` prop (default `'value'`); removes tooltip; maintains precise rendering behavior.
>   - **`AssetsTable`**: Uses `FiatValue` with `mode="price"` for the price column.
>   - **Tests**: Update `FiatValue` tests to assert full numeric formatting and removal of tooltip-dependent behavior.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0ab0ab842eff895a4fcb2605ab7e2c9d69d76ada. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: apps/web/src/components/common/FiatValue/index.tsx [hunk 1]
```diff
@@ -40,23 +42,21 @@ const FiatValue = ({
   }
 
   return (
-    <Tooltip title={precise ? undefined : preciseFiat}>
```
Comment (gfendres) [inline apps/web/src/components/common/FiatValue/index.tsx:43]: it removed the tooltip you know why?

Comment (devpramoth) [reply apps/web/src/components/common/FiatValue/index.tsx:43]: Yes, I added two more additional requirement to match other wallets like Rabby, ledger etc. 

- Remove compact notation (K, M, B suffixes) - always display full numbers

- Remove hover tooltips from FiatValue component

File: packages/utils/src/utils/formatNumber.ts [hunk 2]
```diff
@@ -71,17 +71,80 @@ export const formatAmountPrecise = (number: string | number, precision?: number)
...
+    }
+
+    if (Math.abs(float) < 0.01) {
+      // Get currency symbol for threshold display
+      const zeroFormatted = getCurrencyFormatter(currency, false, 2, 2).format(0)
+      const currencySymbol = zeroFormatted.match(/^(\D+)/)?.[1] || ''
+      const thresholdValue = getCurrencyFormatter(currency, false, 2, 2).format(0.01)
+      // Extract just the number part (0.01) from the formatted string, preserving hair space
+      const numberPart = thresholdValue.replace(/^(\D+)/, '').trim()
+      return `<${currencySymbol} ${numberPart}`
```
[the remark this answers is not in the document]
Comment (devpramoth) [reply packages/utils/src/utils/formatNumber.ts:99]: there cannot be negative prices and values of assets. 0 is the smallest unit.

File: packages/utils/src/utils/formatNumber.ts [hunk 3]
```diff
@@ -71,17 +71,80 @@ export const formatAmountPrecise = (number: string | number, precision?: number)
...
+    }
```
[the remark this answers is not in the document]
Comment (devpramoth) [reply packages/utils/src/utils/formatNumber.ts:100]: there cannot be negative prices and values of assets. 0 is the smallest unit.
